kvbc/tools/object_store_utility/integrity_checker.cpp

```diff
@@ -75,15 +75,14 @@ std::pair<BlockId, Digest> IntegrityChecker::getLatestsCheckpointDescriptor() co
   LOG_INFO(logger_, "Last block: " << lastBlock);
   auto it = dynamic_cast<s3::Client*>(s3_dbset_.metadataDBClient.get())
                 ->getIterator<s3::Client::SortByModifiedDescIterator>();
-  auto [key, val] = it->seekAtLeast(Sliver::copy(checkpoints_prefix_.data(), checkpoints_prefix_.length()));
-  UNUSED(val);
+  const auto& kvdata = it->seekAtLeast(Sliver::copy(checkpoints_prefix_.data(), checkpoints_prefix_.length()));
   if (it->isEnd()) throw std::runtime_error("no checkpoints information in S3 storage");
 
-  std::string suff = key.toString().substr(checkpoints_prefix_.length());
+  std::string suff = kvdata.first.toString().substr(checkpoints_prefix_.length());
```

Comment:
Why creating a new string (allocations, copies)?
 use std::string_view

Reply:
Not recommended to use `string_view`. Changing to `string_view` requires specialized templated function's parameter to be changed to `string_view` from `string`.
https://github.com/vmware/concord-bft/blob/d56ae0e6d8c488a1086105e24f7916fb32c86ec9/util/include/string.hpp#L57 

Inside the function, standard string conversion function is used (`std::stoul(s);`) which again asks for string as parameter. I see no point of doing conversion from `string_view` to `string` again at this point.